src/Microsoft.PowerShell.Commands.Utility/commands/utility/NewTemporaryFileCommand.cs

```diff
@@ -0,0 +1,46 @@
+
+using System;
+using System.IO;
+using System.Management.Automation;
+
+namespace Microsoft.PowerShell.Commands
+{
+    /// <summary>
+    /// The implementation of the "New-TemporaryFile" cmdlet
+    /// </summary>    
+    [Cmdlet(VerbsCommon.New, "TemporaryFile", SupportsShouldProcess = true, HelpUri = "https://go.microsoft.com/fwlink/?LinkId=526726")]
+    [OutputType(typeof(System.IO.FileInfo))]
+    public class NewTemporaryFileCommand : Cmdlet
+    {      
+        /// <summary>
+        /// returns a TemporaryFile
+        /// </summary>
+        protected override void BeginProcessing()
```

Comment:
Putting the logic of a simple cmdlet in the `end` block does protect against stupid pipelines, e.g.

```powershell
function f { begin { write-host 'in begin f'}}
function g { end { write-host 'in end g'}}
g | f
```

This will print
```
in begin f
in end g
```
which is a little weird that the logic for a downstream cmdlet finishes before the upstream command (note no objects are being piped).

In others, you could more easily write some obfuscated (as in, hard to figure out the real logic) pipelines if you use `begin` instead of `end`.